test/aqua.jl

```diff
@@ -22,7 +22,7 @@ using Aqua
     for method_ambiguity in ambs
         @show method_ambiguity
     end
-    @test length(ambs) ≤ 16
+    @test length(ambs) ≤ 13
```

Comment:
this is failing https://github.com/CliMA/ClimaCore.jl/actions/runs/6214646588/job/16866787715?pr=1454#step:7:686
```
Aqua tests (performance): Test Failed at /home/runner/work/ClimaCore.jl/ClimaCore.jl/test/aqua.jl:25
  Expression: length(ambs) ≤ 13
   Evaluated: 15 ≤ 13
```

Reply:
Ah, interesting, it was passing for me locally. I can bump it back up. @simonbyrne, can you re-review the functionality aspect I introduced here?